cmd/systray/systray.go

```diff
@@ -225,3 +268,45 @@ func enableLoggingToConsole() {
 	logger, _ := seelog.LoggerFromConfigAsBytes([]byte(seeConfig))
 	log.ReplaceLogger(logger)
 }
+
+// execCmdOrElevate carries out a command. If current process is not elevated and is not supposely to be elevated, it will launch
+// itself as elevated and quit from the current instance.
+func execCmdOrElevate(cmd string) {
+	if !launchelev && !isUserAdmin {
+		// If not launched as elevated and user is not admin, relaunch self. Use AND here to prevent from dead loop.
+		relaunchElevated(cmd)
+
+		// If elevation failed, just quit to the caller.
+		return
+	}
+
+	if cmds[cmd] != nil {
+		cmds[cmd]()
+	}
+}
+
+// relaunchElevated launch another instance of the current process asking it to carry out a command as admin.
+// If the function succeeds, it will quit the process, otherwise the function will return to the caller.
+func relaunchElevated(cmd string) {
+	verb := "runas"
+	exe, _ := os.Executable()
+	cwd, _ := os.Getwd()
+
+	// Reconstruct arguments, drop launch-gui and tell the new process it should have been elevated.
+	xargs := []string{"-launch-elev=true", "-launch-cmd=" + cmd}
+	args := strings.Join(xargs, " ")
+
+	verbPtr, _ := syscall.UTF16PtrFromString(verb)
```

Comment:
nit: I think I'd prefer we used the `golang.org/x/sys/windows` package instead and call `windows.UTF16PtrFromString()`, instead. Feel free to ignore, but it would be more consistent with other windows places where we manipulate strings.

Reply:
Thanks, will fix in another PR (coming this week)